### ISSUE TITLE
[Bug]: CarPlay not working on M1 Chip - Application does not implement CarPlay template application lifecycle methods in its scene delegate.

### ISSUE THREAD

**Issue by oguzhnatly** [mapbox/mapbox-navigation-ios#3355]
### Mapbox Navigation SDK version

main - https://github.com/mapbox/mapbox-navigation-ios/tree/66e4dd9124c02aa5b729fc72bbe97b44c0b5e747

### Steps to reproduce

1. Required: Apple Silicon M1 Chip
2. Run Example-CarPlay App from MapboxNavigation-SPM.
3. Test with iPhone 11 and iPhone 12 Pro. (Other devices did not tested)

This problem occurred with the M1 chip using Rosetta 2. This is not able to work without Rosetta 2 because of the packages.

### Expected behavior

Example-CarPlay app must run without "Application does not implement CarPlay template application lifecycle methods in its scene delegate." error.

### Actual behavior

When CarPlay is connected, the app crashes immediately.

All errors:
```
2021-09-10 20:09:09.507957+0300 Example[65459:2363953] Failed to load recent items with error: The file “RecentItems.data” couldn’t be opened because there is no such file.
2021-09-10 20:09:09.517082+0300 Example[65459:2363953] Versions of linked Mapbox frameworks:
2021-09-10 20:09:09.565099+0300 Example[65459:2363953] com.mapbox.navigator: 65.0.2
2021-09-10 20:09:09.583976+0300 Example[65459:2363953] com.mapbox.maps: 0.0.1
2021-09-10 20:09:09.599114+0300 Example[65459:2363953] com.mapbox.common: 17.1.0
2021-09-10 20:09:09.611272+0300 Example[65459:2363953] com.mapbox.MapboxMobileEvents: 1.0.2
2021-09-10 20:09:09.640971+0300 Example[65459:2363953] Failed to load recent items with error: The file “RecentItems.data” couldn’t be opened because there is no such file.
2021-09-10 20:09:09.929886+0300 Example[65459:2363953] Metal GPU Frame Capture Enabled
2021-09-10 20:09:09.930776+0300 Example[65459:2363953] Metal API Validation Enabled
2021-09-10 20:09:10.172475+0300 Example[65459:2363953] <maps> Using Metal render backend
2021-09-10 20:09:11.137095+0300 Example[65459:2369498] Connection 1: unable to determine interface type without an established connection
2021-09-10 20:09:11.137166+0300 Example[65459:2369498] Connection 1: unable to determine fallback status without a connection
2021-09-10 20:09:11.207337+0300 Example[65459:2369498] [] nw_protocol_get_quic_image_block_invoke dlopen libquic failed
2021-09-10 20:09:12.354860+0300 Example[65459:2363953] fopen failed for data file: errno = 2 (No such file or directory)
2021-09-10 20:09:12.355064+0300 Example[65459:2363953] Errors found! Invalidating cache...
2021-09-10 20:09:12.514622+0300 Example[65459:2363953] fopen failed for data file: errno = 2 (No such file or directory)
2021-09-10 20:09:12.514726+0300 Example[65459:2363953] Errors found! Invalidating cache...
Failed with error: Error Domain=kCLErrorDomain Code=1 "(null)"
2021-09-10 20:09:13.250962+0300 Example[65459:2363953] <nav-native> PersistentConfig path: /Users/oguzhan/Library/Developer/CoreSimulator/Devices/C03E0E70-9CC3-4C92-9564-D0765329B844/data/Containers/Data/Application/AE778CC4-7C12-4553-BCFD-EF27E7E5FAD6/Library/Application Support/.mapbox/tile_store/navigation/config.json
2021/09/10 17:09:13.261095 [33;1m[WARN][0m No metadata found for tiles
2021/09/10 17:09:13.264294 [33;1m[WARN][0m No metadata found for tiles
2021-09-10 20:09:13.267283+0300 Example[65459:2363953] <nav-native> Tiles path = "/Users/oguzhan/Library/Developer/CoreSimulator/Devices/C03E0E70-9CC3-4C92-9564-D0765329B844/data/Containers/Data/Application/AE778CC4-7C12-4553-BCFD-EF27E7E5FAD6/Library/Application Support/.mapbox/tile_store"
2021-09-10 20:09:13.267861+0300 Example[65459:2363953] <nav-native> TileStore instance was passed
2021-09-10 20:09:13.269357+0300 Example[65459:2363953] <nav-native> Endpoint config = "{"host":"https://api.mapbox.com","isFallback":false,"minDiffInDaysToConsiderServerVersion":0,"navigatorVersion":"65.0.2","token":"***...*","userAgent":"Example/2.0.0 MapboxNavigation/2.0.0 MapboxCoreNavigation/2.0.0 iOS/14.5.0 (x86_64)","version":"","versionBeforeFallback":""}"
2021-09-10 20:09:13.269694+0300 Example[65459:2363953] <nav-native> Creating navigator v.65.0.2
2021/09/10 17:09:13.270200 [33;1m[WARN][0m No metadata found for tiles
2021/09/10 17:09:13.276774 [33;1m[WARN][0m No metadata found for tiles
2021/09/10 17:09:13.277443 [33;1m[WARN][0m No metadata found for tiles
2021/09/10 17:09:13.277768 [33;1m[WARN][0m No metadata found for tiles
2021/09/10 17:09:13.278215 [33;1m[WARN][0m No metadata found for tiles
2021-09-10 20:09:13.314849+0300 Example[65459:2363953] [plugin] AddInstanceForFactory: No factory registered for id <CFUUID 0x6000035cf5e0> F8BB1C28-BAE8-11D6-9C31-00039315CD46
2021-09-10 20:09:13.351744+0300 Example[65459:2363953] <nav-native> Observer callback start
2021-09-10 20:09:13.353606+0300 Example[65459:2363953] <nav-native> Observer callback end
2021-09-10 20:09:13.353674+0300 Example[65459:2363953] <nav-native> Observer callback start
2021-09-10 20:09:13.353739+0300 Example[65459:2363953] <nav-native> Observer callback end
2021-09-10 20:09:13.618958+0300 Example[65459:2370802] <nav-native> PersistentConfig loading started
2021-09-10 20:09:13.619235+0300 Example[65459:2370802] <nav-native> No PersistentConfig found at /Users/oguzhan/Library/Developer/CoreSimulator/Devices/C03E0E70-9CC3-4C92-9564-D0765329B844/data/Containers/Data/Application/AE778CC4-7C12-4553-BCFD-EF27E7E5FAD6/Library/Application Support/.mapbox/tile_store/navigation/config.json
2021-09-10 20:09:13.619355+0300 Example[65459:2370802] <nav-native> PersistentConfig loaded
2021-09-10 20:09:13.619858+0300 Example[65459:2370802] <nav-native> Async config init done with version 2021_09_05-03_00_00
2021-09-10 20:09:13.622415+0300 Example[65459:2370802] <nav-native> PersistentConfig saving started
2021-09-10 20:09:13.626380+0300 Example[65459:2370802] <nav-native> PersistentConfig saved
2021-09-10 20:09:13.644170+0300 Example[65459:2363953] <nav-native> Observer callback start
2021-09-10 20:09:13.644333+0300 Example[65459:2363953] <nav-native> Observer callback end
2021-09-10 20:09:14.740716+0300 Example[65459:2363953] <nav-native> Observer callback start
2021-09-10 20:09:14.740935+0300 Example[65459:2363953] <nav-native> Observer callback end
2021-09-10 20:09:16.056258+0300 Example[65459:2363953] <nav-native> Observer callback start
2021-09-10 20:09:16.056431+0300 Example[65459:2363953] <nav-native> Observer callback end
2021-09-10 20:09:16.659170+0300 Example[65459:2363953] <nav-native> Observer callback start
2021-09-10 20:09:16.659413+0300 Example[65459:2363953] <nav-native> Observer callback end
2021-09-10 20:09:17.693423+0300 Example[65459:2363953] <nav-native> Observer callback start
2021-09-10 20:09:17.693709+0300 Example[65459:2363953] <nav-native> Observer callback end
2021-09-10 20:09:18.721795+0300 Example[65459:2363953] <nav-native> Observer callback start
2021-09-10 20:09:18.722062+0300 Example[65459:2363953] <nav-native> Observer callback end
2021-09-10 20:09:19.773919+0300 Example[65459:2363953] <nav-native> Observer callback start
2021-09-10 20:09:19.774126+0300 Example[65459:2363953] <nav-native> Observer callback end
2021-09-10 20:09:20.780065+0300 Example[65459:2363953] <nav-native> Observer callback start
2021-09-10 20:09:20.780239+0300 Example[65459:2363953] <nav-native> Observer callback end
2021-09-10 20:09:21.812270+0300 Example[65459:2363953] <nav-native> Observer callback start
2021-09-10 20:09:21.812445+0300 Example[65459:2363953] <nav-native> Observer callback end
2021-09-10 20:09:22.797009+0300 Example[65459:2363953] <nav-native> Observer callback start
2021-09-10 20:09:22.797437+0300 Example[65459:2363953] <nav-native> Observer callback end
2021-09-10 20:09:24.205800+0300 Example[65459:2363953] <nav-native> Observer callback start
2021-09-10 20:09:24.205963+0300 Example[65459:2363953] <nav-native> Observer callback end
2021-09-10 20:09:24.212685+0300 Example[65459:2376058] [General] Provided status bar edge is invalid, assigning default
2021-09-10 20:09:24.215716+0300 Example[65459:2363953] [General] Provided status bar edge is invalid, assigning default
2021-09-10 20:09:24.229668+0300 Example[65459:2363953] [db] _LSSchemaConfigureForStore failed with error Error Domain=NSOSStatusErrorDomain Code=-10817 "(null)" UserInfo={_LSFunction=_LSSchemaConfigureForStore, ExpectedSimulatorHash={length = 32, bytes = 0x15dde658 ed2a1267 ab2496d7 34f186ad ... ec431c65 02d68f35 }, _LSLine=409, WrongSimulatorHash={length = 32, bytes = 0xaf25dda9 e45baa35 610eaabd 5bc09901 ... 9cbe61f3 81d7b9d9 }}
2021-09-10 20:09:24.230276+0300 Example[65459:2363953] [db] Failed to initialize client context with error Error Domain=NSOSStatusErrorDomain Code=-10817 "(null)" UserInfo={_LSFunction=_LSSchemaConfigureForStore, ExpectedSimulatorHash={length = 32, bytes = 0x15dde658 ed2a1267 ab2496d7 34f186ad ... ec431c65 02d68f35 }, _LSLine=409, WrongSimulatorHash={length = 32, bytes = 0xaf25dda9 e45baa35 610eaabd 5bc09901 ... 9cbe61f3 81d7b9d9 }}
2021-09-10 20:09:24.232750+0300 Example[65459:2363953] [db] _LSSchemaConfigureForStore failed with error Error Domain=NSOSStatusErrorDomain Code=-10817 "(null)" UserInfo={_LSFunction=_LSSchemaConfigureForStore, ExpectedSimulatorHash={length = 32, bytes = 0x15dde658 ed2a1267 ab2496d7 34f186ad ... ec431c65 02d68f35 }, _LSLine=409, WrongSimulatorHash={length = 32, bytes = 0xaf25dda9 e45baa35 610eaabd 5bc09901 ... 9cbe61f3 81d7b9d9 }}
2021-09-10 20:09:24.232885+0300 Example[65459:2363953] [db] Failed to initialize client context with error Error Domain=NSOSStatusErrorDomain Code=-10817 "(null)" UserInfo={_LSFunction=_LSSchemaConfigureForStore, ExpectedSimulatorHash={length = 32, bytes = 0x15dde658 ed2a1267 ab2496d7 34f186ad ... ec431c65 02d68f35 }, _LSLine=409, WrongSimulatorHash={length = 32, bytes = 0xaf25dda9 e45baa35 610eaabd 5bc09901 ... 9cbe61f3 81d7b9d9 }}
2021-09-10 20:09:24.233533+0300 Example[65459:2363953] [db] _LSSchemaConfigureForStore failed with error Error Domain=NSOSStatusErrorDomain Code=-10817 "(null)" UserInfo={_LSFunction=_LSSchemaConfigureForStore, ExpectedSimulatorHash={length = 32, bytes = 0x15dde658 ed2a1267 ab2496d7 34f186ad ... ec431c65 02d68f35 }, _LSLine=409, WrongSimulatorHash={length = 32, bytes = 0xaf25dda9 e45baa35 610eaabd 5bc09901 ... 9cbe61f3 81d7b9d9 }}
2021-09-10 20:09:24.233626+0300 Example[65459:2363953] [db] Failed to initialize client context with error Error Domain=NSOSStatusErrorDomain Code=-10817 "(null)" UserInfo={_LSFunction=_LSSchemaConfigureForStore, ExpectedSimulatorHash={length = 32, bytes = 0x15dde658 ed2a1267 ab2496d7 34f186ad ... ec431c65 02d68f35 }, _LSLine=409, WrongSimulatorHash={length = 32, bytes = 0xaf25dda9 e45baa35 610eaabd 5bc09901 ... 9cbe61f3 81d7b9d9 }}
2021-09-10 20:09:24.234064+0300 Example[65459:2363953] [db] _LSSchemaConfigureForStore failed with error Error Domain=NSOSStatusErrorDomain Code=-10817 "(null)" UserInfo={_LSFunction=_LSSchemaConfigureForStore, ExpectedSimulatorHash={length = 32, bytes = 0x15dde658 ed2a1267 ab2496d7 34f186ad ... ec431c65 02d68f35 }, _LSLine=409, WrongSimulatorHash={length = 32, bytes = 0xaf25dda9 e45baa35 610eaabd 5bc09901 ... 9cbe61f3 81d7b9d9 }}
2021-09-10 20:09:24.234159+0300 Example[65459:2363953] [db] Failed to initialize client context with error Error Domain=NSOSStatusErrorDomain Code=-10817 "(null)" UserInfo={_LSFunction=_LSSchemaConfigureForStore, ExpectedSimulatorHash={length = 32, bytes = 0x15dde658 ed2a1267 ab2496d7 34f186ad ... ec431c65 02d68f35 }, _LSLine=409, WrongSimulatorHash={length = 32, bytes = 0xaf25dda9 e45baa35 610eaabd 5bc09901 ... 9cbe61f3 81d7b9d9 }}
2021-09-10 20:09:24.234636+0300 Example[65459:2363953] [db] _LSSchemaConfigureForStore failed with error Error Domain=NSOSStatusErrorDomain Code=-10817 "(null)" UserInfo={_LSFunction=_LSSchemaConfigureForStore, ExpectedSimulatorHash={length = 32, bytes = 0x15dde658 ed2a1267 ab2496d7 34f186ad ... ec431c65 02d68f35 }, _LSLine=409, WrongSimulatorHash={length = 32, bytes = 0xaf25dda9 e45baa35 610eaabd 5bc09901 ... 9cbe61f3 81d7b9d9 }}
2021-09-10 20:09:24.234738+0300 Example[65459:2363953] [db] Failed to initialize client context with error Error Domain=NSOSStatusErrorDomain Code=-10817 "(null)" UserInfo={_LSFunction=_LSSchemaConfigureForStore, ExpectedSimulatorHash={length = 32, bytes = 0x15dde658 ed2a1267 ab2496d7 34f186ad ... ec431c65 02d68f35 }, _LSLine=409, WrongSimulatorHash={length = 32, bytes = 0xaf25dda9 e45baa35 610eaabd 5bc09901 ... 9cbe61f3 81d7b9d9 }}
2021-09-10 20:09:24.235152+0300 Example[65459:2363953] [db] _LSSchemaConfigureForStore failed with error Error Domain=NSOSStatusErrorDomain Code=-10817 "(null)" UserInfo={_LSFunction=_LSSchemaConfigureForStore, ExpectedSimulatorHash={length = 32, bytes = 0x15dde658 ed2a1267 ab2496d7 34f186ad ... ec431c65 02d68f35 }, _LSLine=409, WrongSimulatorHash={length = 32, bytes = 0xaf25dda9 e45baa35 610eaabd 5bc09901 ... 9cbe61f3 81d7b9d9 }}
2021-09-10 20:09:24.235282+0300 Example[65459:2363953] [db] Failed to initialize client context with error Error Domain=NSOSStatusErrorDomain Code=-10817 "(null)" UserInfo={_LSFunction=_LSSchemaConfigureForStore, ExpectedSimulatorHash={length = 32, bytes = 0x15dde658 ed2a1267 ab2496d7 34f186ad ... ec431c65 02d68f35 }, _LSLine=409, WrongSimulatorHash={length = 32, bytes = 0xaf25dda9 e45baa35 610eaabd 5bc09901 ... 9cbe61f3 81d7b9d9 }}
2021-09-10 20:09:24.235801+0300 Example[65459:2363953] [db] _LSSchemaConfigureForStore failed with error Error Domain=NSOSStatusErrorDomain Code=-10817 "(null)" UserInfo={_LSFunction=_LSSchemaConfigureForStore, ExpectedSimulatorHash={length = 32, bytes = 0x15dde658 ed2a1267 ab2496d7 34f186ad ... ec431c65 02d68f35 }, _LSLine=409, WrongSimulatorHash={length = 32, bytes = 0xaf25dda9 e45baa35 610eaabd 5bc09901 ... 9cbe61f3 81d7b9d9 }}
2021-09-10 20:09:24.235906+0300 Example[65459:2363953] [db] Failed to initialize client context with error Error Domain=NSOSStatusErrorDomain Code=-10817 "(null)" UserInfo={_LSFunction=_LSSchemaConfigureForStore, ExpectedSimulatorHash={length = 32, bytes = 0x15dde658 ed2a1267 ab2496d7 34f186ad ... ec431c65 02d68f35 }, _LSLine=409, WrongSimulatorHash={length = 32, bytes = 0xaf25dda9 e45baa35 610eaabd 5bc09901 ... 9cbe61f3 81d7b9d9 }}
2021-09-10 20:09:24.236465+0300 Example[65459:2363953] [db] _LSSchemaConfigureForStore failed with error Error Domain=NSOSStatusErrorDomain Code=-10817 "(null)" UserInfo={_LSFunction=_LSSchemaConfigureForStore, ExpectedSimulatorHash={length = 32, bytes = 0x15dde658 ed2a1267 ab2496d7 34f186ad ... ec431c65 02d68f35 }, _LSLine=409, WrongSimulatorHash={length = 32, bytes = 0xaf25dda9 e45baa35 610eaabd 5bc09901 ... 9cbe61f3 81d7b9d9 }}
2021-09-10 20:09:24.236562+0300 Example[65459:2363953] [db] Failed to initialize client context with error Error Domain=NSOSStatusErrorDomain Code=-10817 "(null)" UserInfo={_LSFunction=_LSSchemaConfigureForStore, ExpectedSimulatorHash={length = 32, bytes = 0x15dde658 ed2a1267 ab2496d7 34f186ad ... ec431c65 02d68f35 }, _LSLine=409, WrongSimulatorHash={length = 32, bytes = 0xaf25dda9 e45baa35 610eaabd 5bc09901 ... 9cbe61f3 81d7b9d9 }}
2021-09-10 20:09:24.237062+0300 Example[65459:2363953] [db] _LSSchemaConfigureForStore failed with error Error Domain=NSOSStatusErrorDomain Code=-10817 "(null)" UserInfo={_LSFunction=_LSSchemaConfigureForStore, ExpectedSimulatorHash={length = 32, bytes = 0x15dde658 ed2a1267 ab2496d7 34f186ad ... ec431c65 02d68f35 }, _LSLine=409, WrongSimulatorHash={length = 32, bytes = 0xaf25dda9 e45baa35 610eaabd 5bc09901 ... 9cbe61f3 81d7b9d9 }}
2021-09-10 20:09:24.237147+0300 Example[65459:2363953] [db] Failed to initialize client context with error Error Domain=NSOSStatusErrorDomain Code=-10817 "(null)" UserInfo={_LSFunction=_LSSchemaConfigureForStore, ExpectedSimulatorHash={length = 32, bytes = 0x15dde658 ed2a1267 ab2496d7 34f186ad ... ec431c65 02d68f35 }, _LSLine=409, WrongSimulatorHash={length = 32, bytes = 0xaf25dda9 e45baa35 610eaabd 5bc09901 ... 9cbe61f3 81d7b9d9 }}
2021-09-10 20:09:24.237397+0300 Example[65459:2363953] [General] Not attempting endpoint connection.
2021-09-10 20:09:24.238034+0300 Example[65459:2363953] [db] _LSSchemaConfigureForStore failed with error Error Domain=NSOSStatusErrorDomain Code=-10817 "(null)" UserInfo={_LSFunction=_LSSchemaConfigureForStore, ExpectedSimulatorHash={length = 32, bytes = 0x15dde658 ed2a1267 ab2496d7 34f186ad ... ec431c65 02d68f35 }, _LSLine=409, WrongSimulatorHash={length = 32, bytes = 0xaf25dda9 e45baa35 610eaabd 5bc09901 ... 9cbe61f3 81d7b9d9 }}
2021-09-10 20:09:24.238121+0300 Example[65459:2363953] [db] Failed to initialize client context with error Error Domain=NSOSStatusErrorDomain Code=-10817 "(null)" UserInfo={_LSFunction=_LSSchemaConfigureForStore, ExpectedSimulatorHash={length = 32, bytes = 0x15dde658 ed2a1267 ab2496d7 34f186ad ... ec431c65 02d68f35 }, _LSLine=409, WrongSimulatorHash={length = 32, bytes = 0xaf25dda9 e45baa35 610eaabd 5bc09901 ... 9cbe61f3 81d7b9d9 }}
2021-09-10 20:09:24.238550+0300 Example[65459:2363953] [db] _LSSchemaConfigureForStore failed with error Error Domain=NSOSStatusErrorDomain Code=-10817 "(null)" UserInfo={_LSFunction=_LSSchemaConfigureForStore, ExpectedSimulatorHash={length = 32, bytes = 0x15dde658 ed2a1267 ab2496d7 34f186ad ... ec431c65 02d68f35 }, _LSLine=409, WrongSimulatorHash={length = 32, bytes = 0xaf25dda9 e45baa35 610eaabd 5bc09901 ... 9cbe61f3 81d7b9d9 }}
2021-09-10 20:09:24.238643+0300 Example[65459:2363953] [db] Failed to initialize client context with error Error Domain=NSOSStatusErrorDomain Code=-10817 "(null)" UserInfo={_LSFunction=_LSSchemaConfigureForStore, ExpectedSimulatorHash={length = 32, bytes = 0x15dde658 ed2a1267 ab2496d7 34f186ad ... ec431c65 02d68f35 }, _LSLine=409, WrongSimulatorHash={length = 32, bytes = 0xaf25dda9 e45baa35 610eaabd 5bc09901 ... 9cbe61f3 81d7b9d9 }}
2021-09-10 20:09:24.265590+0300 Example[65459:2363953] *** Terminating app due to uncaught exception 'NSGenericException', reason: 'Application does not implement CarPlay template application lifecycle methods in its scene delegate.'
*** First throw call stack:
(
	0   CoreFoundation                      0x00000001110fafba __exceptionPreprocess + 242
	1   libobjc.A.dylib                     0x0000000110bc6ff5 objc_exception_throw + 48
	2   CoreFoundation                      0x00000001110fae98 -[NSException initWithCoder:] + 0
	3   CarPlay                             0x0000000110c3cd57 -[CPTemplateApplicationScene _deliverInterfaceControllerToDelegate] + 860
	4   CarPlay                             0x0000000110c3b9bb __64-[CPTemplateApplicationScene initWithSession:connectionOptions:]_block_invoke.23 + 125
	5   Foundation                          0x00000001106d77ec -[__NSObserver _doit:] + 287
	6   CoreFoundation                      0x000000011103b882 __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
	7   CoreFoundation                      0x000000011103b851 ___CFXRegistrationPost_block_invoke + 49
	8   CoreFoundation                      0x000000011103af81 _CFXRegistrationPost + 454
	9   CoreFoundation                      0x000000011103a8ef _CFXNotificationPost + 796
	10  Foundation                          0x00000001106d8027 -[NSNotificationCenter postNotificationName:object:userInfo:] + 59
	11  UIKitCore                           0x000000011c0497f2 +[UIScene _sceneForFBSScene:create:withSession:connectionOptions:] + 1645
	12  UIKitCore                           0x000000011cbfddf9 -[UIApplication _connectUISceneFromFBSScene:transitionContext:] + 1114
	13  UIKitCore                           0x000000011cbfe128 -[UIApplication workspace:didCreateScene:withTransitionContext:completion:] + 289
	14  UIKitCore                           0x000000011c6dfab4 -[UIApplicationSceneClientAgent scene:didInitializeWithEvent:completion:] + 358
	15  FrontBoardServices                  0x000000012118740b -[FBSScene _callOutQueue_agent_didCreateWithTransitionContext:completion:] + 398
	16  FrontBoardServices                  0x00000001211afe55 __94-[FBSWorkspaceScenesClient createWithSceneID:groupID:parameters:transitionContext:completion:]_block_invoke.176 + 102
	17  FrontBoardServices                  0x0000000121194f12 -[FBSWorkspace _calloutQueue_executeCalloutFromSource:withBlock:] + 209
	18  FrontBoardServices                  0x00000001211afb28 __94-[FBSWorkspaceScenesClient createWithSceneID:groupID:parameters:transitionContext:completion:]_block_invoke + 352
	19  libdispatch.dylib                   0x00000001146ce74e _dispatch_client_callout + 8
	20  libdispatch.dylib                   0x00000001146d1656 _dispatch_block_invoke_direct + 295
	21  FrontBoardServices                  0x00000001211d55d0 __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__ + 30
	22  FrontBoardServices                  0x00000001211d52b6 -[FBSSerialQueue _targetQueue_performNextIfPossible] + 433
	23  FrontBoardServices                  0x00000001211d577b -[FBSSerialQueue _performNextFromRunLoopSource] + 22
	24  CoreFoundation                      0x0000000111068ede __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
	25  CoreFoundation                      0x0000000111068dd6 __CFRunLoopDoSource0 + 180
	26  CoreFoundation                      0x0000000111068300 __CFRunLoopDoSources0 + 340
	27  CoreFoundation                      0x00000001110629f7 __CFRunLoopRun + 875
	28  CoreFoundation                      0x00000001110621a7 CFRunLoopRunSpecific + 567
	29  GraphicsServices                    0x00000001213c2d85 GSEventRunModal + 139
	30  UIKitCore                           0x000000011cbfc4df -[UIApplication _run] + 912
	31  UIKitCore                           0x000000011cc0139c UIApplicationMain + 101
	32  Example                             0x0000000104d6f99b main + 75
	33  libdyld.dylib                       0x000000011475cbbd start + 1
	34  ???                                 0x0000000000000001 0x0 + 1
)
libc++abi: terminating with uncaught exception of type NSException
*** Terminating app due to uncaught exception 'NSGenericException', reason: 'Application does not implement CarPlay template application lifecycle methods in its scene delegate.'
CoreSimulator 775 - Device: iPhone 11 (C03E0E70-9CC3-4C92-9564-D0765329B844) - Runtime: iOS 14.5 (18E182) - DeviceType: iPhone 11
terminating with uncaught exception of type NSException
_LSContextInitReturningError() failed with error Error Domain=NSOSStatusErrorDomain Code=-10817 "(null)" UserInfo={_LSFunction=_LSSchemaConfigureForStore, ExpectedSimulatorHash={length = 32, bytes = 0x15dde658 ed2a1267 ab2496d7 34f186ad ... ec431c65 02d68f35 }, _LSLine=409, WrongSimulatorHash={length = 32, bytes = 0xaf25dda9 e45baa35 610eaabd 5bc09901 ... 9cbe61f3 81d7b9d9 }}
```

### Is this a one-time issue or a repeatable issue?

repeatable

**Comment by S2Ler** [comment]
@oguzhnatly Hi, thanks for reporting. Couple of clarification questions:
- What Xcode version do you use?
- Do you run Xcode in Rosetta 2 or just simulator?
- Do you use Rosetta 2 on M1 because of the issue with our SDK or because of some other SDK you use? I'm asking this because our SDK v2 should work on M1 macOS perfectly.

**Comment by oguzhnatly** [comment]
Hi, thanks for the quick reply. Actually, both of them.

1. Using other SDK

In my SDK, my sample carplay application using SDK works perfectly without any errors before the Mapbox implementation, natively on M1 or just using Rosetta 2.

With the Mapbox;
At first, I've tried to natively integrate this Mapbox repository into my example project that uses my SDK on M1, but when I've encountered the x86 64 error, I switched to Rosetta 2 on M1. It works without a carplay simulator, but when the carplay is connected to the iPhone simulator, it crashes immediately. The error is very interesting at this moment, but I got:
```
libc++abi: terminating with uncaught exception of type NSException
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'Unsupported object <CPTabBarTemplate: 0x60000043a9a0> <identifier: C7247644-AE18-4B8D-B960-95DD25544B51, userInfo: (null), tabTitle: (null), tabImage: (null), showsTabBadge: 0> passed to setRootTemplate:animated:completion:. Allowed classes: (null)'
terminating with uncaught exception of type NSException
CoreSimulator 775 - Device: iPhone 12 Pro (193CA20D-65B7-4CBC-9FB8-D4C0CCA78D83) - Runtime: iOS 14.5 (18E182) - DeviceType: iPhone 12 Pro
_LSContextInitReturningError() failed with error Error Domain=NSOSStatusErrorDomain Code=-10817 "(null)" UserInfo={_LSFunction=_LSSchemaConfigureForStore, ExpectedSimulatorHash={length = 32, bytes = 0x15dde658 ed2a1267 ab2496d7 34f186ad ... ec431c65 02d68f35 }, _LSLine=409, WrongSimulatorHash={length = 32, bytes = 0xaf25dda9 e45baa35 610eaabd 5bc09901 ... 9cbe61f3 81d7b9d9 }}
```
Such a thing `Allowed classes: (null)` is almost impossible to happen. Normally, it must contain the name of CarPlay Templates.

2. Without any SDK -- the SPM example project of Mapbox

I've also tried to test the MapboxNavigation-SPM.xcodeproj example project natively on M1 without the use of any SDK or other tools. I simply double-clicked it and ran it. It was crashing when the carplay is connected to the simulator, and I've tried testing it using Rosetta 2 as well, it works without a carplay simulator but when I opened the carplay simulator, it crashes immediately. This time, the error I mentioned in the first comment.

By the way, I've also tested the example project in these cases:
- Xcode using Rosetta 2, Simulator using Rosetta 2
- Xcode using Rosetta 2, Simulator native
- Xcode native, Simulator native

Between all cases, caches are cleared and the project is build again. Unfortunately, it's not working in any case.
Xcode Version: Version 12.5.1 (12E507)

---

**UPDATE:** While I'm writing this comment, I downloaded the main branch from https://github.com/mapbox/mapbox-navigation-ios/tree/03f7871ad4894a41a7bc1595919214404079ced7 and now working on the M1 chip natively. It's highly possible that this is related to some missing files in the MapboxNavigation project. Thanks @MaximAlien!
<img width="1223" alt="Screen Shot 2021-09-11 at 13 58 11" src="https://user-images.githubusercontent.com/54781138/132945554-b58df8fc-dc47-4aca-a130-9568c53e5aa7.png">
